docs/user_guides/single_table/custom_constraints.rst

```diff
@@ -0,0 +1,239 @@
+.. _custom_constraints:
+
+Defining Custom Constraints
+===========================
+
+In some cases, the predefined constraints do not cover all your needs. 
+In such scenarios, you can use ``CustomConstraint`` to define your own 
+logic on how you would like to apply it to your data. There are three 
```

Comment:
More of a writing comment, but I'm not sure what "it" refers to. Maybe say "define your own logic on how to constrain your data"